checkov/sca_package/scanner.py

```diff
@@ -5,116 +5,102 @@
+            for input_path in input_paths:
+                scan_results.append(self.run_scan(input_path))
         else:
-            async with Pool() as pool:
-                scan_results = await pool.starmap(self.run_scan, args)
+            input_paths = [(input_path,) for input_path in input_paths]
+            with Pool() as pool:
+                scan_results = pool.starmap(self.run_scan, input_paths)
 
         return scan_results
 
-    async def run_scan(
-        self,
-        command: str,
-        input_path: Path,
-        output_path: Path,
-    ) -> Dict[str, Any]:
+    def run_scan(self, input_path: Path) -> dict:
         logging.info(f"Start to scan package file {input_path}")
-        process = await asyncio.create_subprocess_shell(
-            command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+
+        request_body = {
+            "compressedFileBody": compress_file_gzip_base64(str(input_path)),
+            "compressionMethod": "gzip",
+            "fileName": input_path.name
+        }
+
+        response = requests.request(
+            "POST", f"{self.base_url}/api/v1/vulnerabilities/scan",
+            headers=bc_integration.get_default_headers("GET"),
+            data=request_body
         )
 
-        stdout, stderr = await process.communicate()
+        response.raise_for_status()
+        response_json = response.json()
+
+        if response_json["status"] == "already_exist":
+            return json.loads(
+                decompress_file_gzip_base64(
+                    response_json["outputData"]
+                )
+            )
 
-        # log output for debugging
-        logging.debug(stdout.decode())
+        return self.run_scan_busy_wait(response_json['id'])
 
-        exit_code = await process.wait()
+    def run_scan_busy_wait(self, scan_id: str) -> dict:
+        current_state = "Empty"
+        desired_state = "Result"
+        total_sleeping_time = 0
+        response = requests.Response()
 
-        if exit_code:
-            logging.error(f"Failed to scan package file {input_path}")
-            logging.error(stderr.decode())
-            return {}
+        while current_state != desired_state:
+            response = requests.request(
+                "GET", f"{self.base_url}/api/v1/vulnerabilities/scan-results/{scan_id}",
+                headers=bc_integration.get_default_headers("GET")
+            )
+            response_json = response.json()
+            current_state = response_json["outputType"]
 
-        logging.info(f"Successfully scanned package file {input_path}")
+            if current_state == "Error":
+                logging.error(response_json["outputData"])
+                return {}
 
-        # read and delete the report file
-        scan_result = json.loads(output_path.read_text())
-        output_path.unlink()
+            time.sleep(SLEEP_DURATION)
+            total_sleeping_time += SLEEP_DURATION
```

Comment:
@Eliran-Turgeman let's put this line after the `if total_sleeping_time > MAX_SLEEP_DURATION:` to avoid waiting without making a call